TestFoundation/Utilities.swift

```diff
@@ -134,50 +132,313 @@ extension Optional {
 }
 
-func expectTrue(_ actual: Bool, _ message: @autoclosure () -> String = "") {
-    XCTAssertTrue(actual, message())
+func expectFalse(_ actual: Bool, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
+    XCTAssertFalse(actual, message(), file: file, line: line)
 }
 
-func expectFalse(_ actual: Bool, _ message: @autoclosure () -> String = "") {
-    XCTAssertFalse(actual, message())
+func expectEqual<T: Equatable>(_ expected: T, _ actual: T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
+    XCTAssertEqual(expected, actual, message(), file: file, line: line)
 }
 
-func expectEqual<T: Equatable>(_ expected: T, _ actual: T, _ message: @autoclosure () -> String = "") {
-    XCTAssertEqual(expected, actual, message())
+func expectNotEqual<T: Equatable>(_ expected: T, _ actual: T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
+    XCTAssertNotEqual(expected, actual, message(), file: file, line: line)
 }
 
-func expectEqual<T: FloatingPoint>(_ expected: T, _ actual: T, within: T, _ message: @autoclosure () -> String = "") {
-    XCTAssertEqual(expected, actual, accuracy: within, message())
+func expectEqual<T: FloatingPoint>(_ expected: T, _ actual: T, within: T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
+    XCTAssertEqual(expected, actual, accuracy: within, message(), file: file, line: line)
 }
 
-func expectEqual<T: FloatingPoint>(_ expected: T?, _ actual: T, within: T, _ message: @autoclosure () -> String = "") {
-    XCTAssertNotNil(expected, message())
+func expectEqual<T: FloatingPoint>(_ expected: T?, _ actual: T, within: T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
+    XCTAssertNotNil(expected, message(), file: file, line: line)
     if let expected = expected {
-        XCTAssertEqual(expected, actual, accuracy: within, message())
+        XCTAssertEqual(expected, actual, accuracy: within, message(), file: file, line: line)
     }
 }
 
+func expectEqual(
+    _ expected: Any.Type,
+    _ actual: Any.Type,
+    _ message: @autoclosure () -> String = "",
+    file: StaticString = #file,
+    line: UInt = #line
+) {
+    XCTAssertTrue(expected == actual, message(), file: file, line: line)
+}
 
 extension Fixture where ValueType: NSObject & NSCoding {
     func loadEach(handler: (ValueType, FixtureVariant) throws -> Void) throws {
         try self.loadEach(fixtureRepository: try testBundle().url(forResource: "Fixtures", withExtension: nil).unwrapped(), handler: handler)
     }
 }
+
+/// Test that the elements of `instances` satisfy the semantic
+/// requirements of `Equatable`, using `oracle` to generate equality
+/// expectations from pairs of positions in `instances`.
+///
+/// - Note: `oracle` is also checked for conformance to the
+///   laws.
+public func checkEquatable<Instances: Collection>(
```

Comment:
Are we okay having a duplicate implementation of this between SCF and the stdlib tests, or is there something we can do?

This does not block the patch, but it's a point of concern going forward.

Reply:
cc @lorentey

Reply:
Oh yes! The XCTest-StdlibUnittest dichotomy is super annoying, we should do something about that. Ideally we would have a single unified testing infrastructure.

The high-level behavioral tests should be shared between all platforms, too.